### PR TITLE
Add spark k8s operator docs

### PR DESCRIPTION
**Pull request by razvan**
Depends on: https://github.com/stackabletech/spark-k8s-operator/pull/41

### CODE REVIEW

**Review by siegfriedweber — CHANGES_REQUESTED** [review]
* The repository must be added to `gitpod-antora-playbook.yml` for the sake of completeness.
* The operator should be added to the components list in [modules/ROOT/pages/index.adoc](https://github.com/stackabletech/documentation/blob/28f82518574e9f149776eb63fa74cce6a4431111/modules/ROOT/pages/index.adoc#components).
* The supported versions should be linked in [modules/ROOT/pages/supported-versions.adoc](https://github.com/stackabletech/documentation/blob/28f82518574e9f149776eb63fa74cce6a4431111/modules/ROOT/pages/supported-versions.adoc#supported-product-versions).
* The repository should be added to [modules/contributor/pages/development_dashboard.adoc](https://github.com/stackabletech/documentation/blob/28f82518574e9f149776eb63fa74cce6a4431111/modules/contributor/pages/development_dashboard.adoc#development-dashboard).
* Optional: The repositories in the playbooks are ordered alphabetically, so `spark-k8s-operator` should be added before `spark-operator`.

File: modules/contributor/pages/development_dashboard.adoc [hunk 1]
```diff
@@ -77,6 +77,15 @@ This page contains information about our products and relevant links you might n
 |https://docs.stackable.tech/spark/index.html
 |
 
+|Apache Spark on Kubernetes
+|https://spark.apache.org/
+|https://github.com/apache/spark
+|https://github.com/stackabletech/spark-k8s-operator
+|https://github.com/stackabletech/docker-images/tree/main/spark-k8s
+|https://ci.stackable.tech/job/Spark%20Operator%20Integration%20Tests/
```
Comment (siegfriedweber) [inline modules/contributor/pages/development_dashboard.adoc:85]: The link to the Jenkins job is not correct. The documentation can either be adapted when the Jenkins job is created or (what I would prefer) the link can be set to the (hopefully) correct URL.

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: modules/contributor/pages/development_dashboard.adoc (hunk 1 above)
Comment (razvan) [reply modules/contributor/pages/development_dashboard.adoc]: Done.

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Pull request merged**